### PR TITLE
Fixes Tinea Luxor

### LINKED ISSUE #4651: Lavaland Chemicals (Tinea Luxor)
## Round ID:
30803

## Testmerges:
#4415: 'Let's see if this helps anything at all.' by francinum at commit 41b343ce2c
#4380: 'Security Rework Part 1 of ?: Energy pistols' by ike709 at commit c10d3a13cb
#4357: 'Ports new overlay lighting component and opacity refactor from tg' by yorii at commit 0f934881d1

## Reproduction:
Take Tinea Luxor, a chemical that is intended to cause glowing
Don't glow
be sad

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

Title. It wasn't glowing before, apparently. Closes #4651 

## Why It's Good For The Game

Fixes bug.

## Changelog
:cl:
fix: Tinea Luxor now actually makes you glow
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
this doesn't make you glow. The bug is that it makes you go when it reacts in side you (I think, though it might be when splashed o you) rather than when you start metabolizing it.

**Comment by PowerfulBacon** [comment]
to fix this change /datum/reagent/consumable/tinlux/reaction_mob(mob/living/M) to on_mob_begin_metabolize or whatever the proc is

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 1]
```diff
@@ -671,11 +671,13 @@
 	color = "#b5a213"
 	taste_description = "tingling mushroom"
 
-/datum/reagent/consumable/tinlux/reaction_mob(mob/living/M)
+/datum/reagent/consumable/tinlux/on_mob_add(mob/living/carbon/M)
+	. = ..()
```
Comment (PowerfulBacon) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]:
```suggestion
/datum/reagent/consumable/tinlux//datum/reagent/proc/on_mob_metabolize(mob/living/L)
(mob/living/carbon/M)
```

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 2]
```diff
@@ -671,11 +671,13 @@
...
 	M.set_light(2)
 
-/datum/reagent/consumable/tinlux/on_mob_end_metabolize(mob/living/M)
-	M.set_light(-2)
+/datum/reagent/consumable/tinlux/on_mob_delete(mob/living/M)
+	. = ..()
```
Comment (PowerfulBacon) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]:
```suggestion
/datum/reagent/consumable/tinlux/on_mob_end_metabolize(mob/living/M)
```

**Review by PowerfulBacon — COMMENTED** [review]
This accounts for having a liver, assuming you want that. Also the base code for on_mob_add does tnothing

**Comment by ivanmixo** [comment]
I believe I tried this, but something was wrong with it. Give me a moment.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 3]
```diff
@@ -671,11 +671,13 @@
 	color = "#b5a213"
 	taste_description = "tingling mushroom"
 
-/datum/reagent/consumable/tinlux/reaction_mob(mob/living/M)
+/datum/reagent/consumable/tinlux/on_mob_metabolize(mob/living/carbon/M)
+	. = ..()
```
Comment (PowerfulBacon) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Unnecessary ..() call, parent proc is empty.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 4]
```diff
@@ -671,11 +671,13 @@
...
 	M.set_light(2)
 
-/datum/reagent/consumable/tinlux/on_mob_end_metabolize(mob/living/M)
-	M.set_light(-2)
+/datum/reagent/consumable/tinlux/on_mob_end_metabolize(mob/living/carbon/M)
+	. = ..()
```
Comment (PowerfulBacon) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Unnecessary ..() call, parent proc is empty.

**Comment by RDS88-beeman** [comment]
wow this chem is fucked

**Comment by ivanmixo** [comment]
Linters failing isn't my fault. Fun. It compiles fine on local.

**Review by utilbird — APPROVED** [review]

**Pull request merged**